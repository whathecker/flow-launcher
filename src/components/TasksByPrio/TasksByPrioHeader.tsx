/* eslint-disable @typescript-eslint/no-unused-vars */
import React from "react";
import { Image } from "react-native";
import { View, Text } from "../Themed";
import { GoBackButtonWhite } from "../shared";
import { Container, Typography, Color } from "../../styles";
import { navigationRef } from "../../utils";
import { PriorityTier } from "../../types/core/value-object";

type TasksByPrioHeaderProps = {
  prio: PriorityTier;
  goalTitle: string; // TODO: consider to use state from task context instead?
  label: string;
  backgroundColor: string;
};

//TODO: consider to use default View for the nested one?
const TasksByPrioHeader: React.FC<TasksByPrioHeaderProps> = ({
  prio,
  goalTitle,
  label,
  backgroundColor,
}: TasksByPrioHeaderProps) => {
  return (
    <View lightColor={backgroundColor} darkColor={backgroundColor}>
      <View
        lightColor={backgroundColor}
        darkColor={backgroundColor}
        style={{
          marginTop: "13%",
          ...Container.flexStart,
          //minHeight: "50%",
        }}
      >
        <View
          lightColor={backgroundColor}
          darkColor={backgroundColor}
          style={{
            paddingLeft: "8%",
            paddingTop: "2.5%",
          }}
        >
          <GoBackButtonWhite
            backgroundColor={backgroundColor}
            pressHandler={() => navigationRef.goBack()}
          />
        </View>
      </View>
      <View
        lightColor={backgroundColor}
        darkColor={backgroundColor}
        style={{
          paddingTop: "7%",
          paddingLeft: "12%",
        }}
      >
        <View
          lightColor={backgroundColor}
          darkColor={backgroundColor}
          style={{ ...Container.flexStart, paddingBottom: "12%" }}
        >
          <Image
            style={{ width: 32, height: 32 }}
            source={require("../../../assets/images/double-exclamation-mark_203c-fe0f.png")}
          />
          <Text
            lightColor={Color.light.textOnColorForRead}
            darkColor={Color.dark.textOnColorForRead}
            style={{
              ...Typography.p,
              fontSize: 18,
              paddingLeft: "3%",
            }}
          >
            {label}
          </Text>
        </View>
        <Text
          lightColor={Color.light.textOnColorForRead}
          darkColor={Color.dark.textOnColorForRead}
          style={{
            ...Typography.h4,
            fontSize: 22,
            paddingBottom: "6%",
          }}
        >
          {goalTitle}
        </Text>
      </View>
    </View>
  );
};

export default TasksByPrioHeader;
