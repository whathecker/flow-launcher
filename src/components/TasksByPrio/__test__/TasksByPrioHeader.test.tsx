import * as React from "react";
import renderer from "react-test-renderer";
import TasksByPrioHeader from "../TasksByPrioHeader";

describe("Test TasksByPrioHeader component", () => {
  it("Should renders TasksByPrioHeader correctly", () => {
    const treeHighest = renderer
      .create(
        <TasksByPrioHeader
          prio="highest"
          goalTitle="Mock goal"
          label={"dummy label"}
          backgroundColor="#BCA136"
        />,
      )
      .toJSON();

    const treeHigh = renderer
      .create(
        <TasksByPrioHeader
          prio="high"
          goalTitle="Mock goal"
          label={"dummy label"}
          backgroundColor="#BCA136"
        />,
      )
      .toJSON();

    const treeMid = renderer
      .create(
        <TasksByPrioHeader
          prio="mid"
          goalTitle="Mock goal"
          label={"dummy label"}
          backgroundColor="#BCA136"
        />,
      )
      .toJSON();

    const treeLow = renderer
      .create(
        <TasksByPrioHeader
          prio="low"
          goalTitle="Mock goal"
          label={"dummy label"}
          backgroundColor="#BCA136"
        />,
      )
      .toJSON();

    expect(treeHighest).toMatchSnapshot();
    expect(treeHigh).toMatchSnapshot();
    expect(treeMid).toMatchSnapshot();
    expect(treeLow).toMatchSnapshot();
  });
});
