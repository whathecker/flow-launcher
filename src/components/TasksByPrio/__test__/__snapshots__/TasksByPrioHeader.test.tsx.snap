// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TasksByPrioHeader component Should renders TasksByPrioHeader correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#BCA136",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#BCA136",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginTop": "13%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#BCA136",
          },
          Object {
            "paddingLeft": "8%",
            "paddingTop": "2.5%",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#BCA136",
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#BCA136",
        },
        Object {
          "paddingLeft": "12%",
          "paddingTop": "7%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#BCA136",
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingBottom": "12%",
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 32,
            "width": 32,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
            },
            Object {
              "fontFamily": "Lexend-Regular",
              "fontSize": 18,
              "paddingLeft": "3%",
            },
          ]
        }
      >
        dummy label
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
          },
          Object {
            "fontFamily": "Lexend-SemiBold",
            "fontSize": 22,
            "paddingBottom": "6%",
          },
        ]
      }
    >
      Mock goal
    </Text>
  </View>
</View>
`;
